cfi/lib/src/xoshiro.rs

```diff
@@ -0,0 +1,235 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    xoshiro.rs
+
+Abstract:
+
+    File contains implementation of RNG based on System.Random Xoshiro128** algorithm.
+
+References:
+    https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro128StarStarImpl.cs
+
+--*/
+
+use crate::{cfi_panic, CfiPanicInfo};
+#[cfg(not(feature = "cfi-test"))]
+use caliptra_common::memory_layout::{CFI_XO_S0_ORG, CFI_XO_S1_ORG, CFI_XO_S2_ORG, CFI_XO_S3_ORG};
+
+#[cfg(feature = "cfi-test")]
+static mut S0: u32 = 0u32;
+#[cfg(feature = "cfi-test")]
+static mut S1: u32 = 0u32;
+#[cfg(feature = "cfi-test")]
+static mut S2: u32 = 0u32;
+#[cfg(feature = "cfi-test")]
+static mut S3: u32 = 0u32;
+
+pub struct Xoshiro128Reg;
+
+impl Xoshiro128Reg {
+    pub fn s0(&self) -> u32 {
+        unsafe {
+            #[cfg(feature = "cfi-test")]
+            {
+                core::ptr::read_volatile(&S0 as *const u32)
```

Comment:
While volatile? These values aren't going to change behind the back of the compiler, are they?

Reply:
Fixed.